evalml/automl/automl_search.py

```diff
@@ -791,8 +803,12 @@ def search(self, show_iteration_plot=True):
             show_iteration_plot (boolean, True): Shows an iteration vs. score plot in Jupyter notebook.
                 Disabled by default in non-Jupyter enviroments.
         """
+        if self.verbose:
+            logger = get_logger(__name__)
+        else:
+            logger = logging.getLogger(__name__)
```

Comment:
Is there a way we can avoid duplicate code? Over time the duplicate paths could drift.

One idea: do
```python
        if self.verbose:
            self.logger = get_logger(__name__)
        else:
            self.logger = logging.getLogger(__name__)
```
in `AutoMLSearch.__init__` as you are already doing. And then in the rest of the `AutoMLSearch` code, i.e. `search` and `_should_continue`, reference `self.logger` instead of `logger`

Reply:
I had that implementation when I started out - the issue that I ran into was with reverting the change afterwards. 

We want to make sure our logging configuration is removed as soon as it's no longer necessary, so that (for example) if you run search with `verbose=True` followed by `verbose=False`, the second run does not log anything. However, simply reverting the logging at the end of search does not guarantee this reversion - a user might not run search after instantiation, and may change the verbose flag in between those runs (I ran into this issue when running our tests!). Therefore to make this as flexible as possible, it makes the most sense to set up and tear down the logger within each function independently, so we don't carry over unwanted logging functionality anywhere.

That being said, I'm more than happy to throw this logic into a helper function to clean up code duplication!

Reply:
Ah right, understood, thanks for explaining that.

Hmm, perhaps this complexity around reversion is an argument for removing the verbosity flag altogether. What do you think, would that be helpful here?